components/org.wso2.carbon.identity.oauth.par/src/main/java/org/wso2/carbon/identity/oauth/par/model/OAuthParRequestWrapper.java

```
@@ -0,0 +1,82 @@
+ * under the License.
+ */
+package org.wso2.carbon.identity.oauth.par.model;
+
+
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.wso2.carbon.identity.oauth.common.OAuthConstants;
+import org.wso2.carbon.identity.oauth.par.common.ParConstants;
+import org.wso2.carbon.identity.oauth.par.core.ParAuthService;
+import org.wso2.carbon.identity.oauth.par.exceptions.ParAuthFailureException;
+import org.wso2.carbon.identity.oauth.par.exceptions.ParCoreException;
+
+import java.util.Map;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletRequestWrapper;
+
+
+/**
+ * Wrapper class to handle PAR request where the parameters of the request will be replaced
+ * with the parameters obtained from the PAR request at the PAR endpoint by accessing the relevant
+ * set of parameters for the submitted request_uri from the store.
+ */
+public class OAuthParRequestWrapper extends HttpServletRequestWrapper {
+
+    private final Map<String, String> params;
+
+    /**
+     * Wraps the request with parameters obtained from the PAR endpoint.
+     */
+    public OAuthParRequestWrapper(HttpServletRequest request, ParAuthService parAuthService)
+            throws OAuthProblemException {
+
+        super(request);
+
+        //get only uuid from request_uri
+        String uuid = request.getParameter(OAuthConstants.OAuth20Params.REQUEST_URI)
+                .replaceFirst(ParConstants.REQUEST_URI_HEAD, "");
+
+        try {
+            if (parAuthService == null) {
+                throw new ParAuthFailureException("ParAuthService is not initialized properly");
+            }
+
+            params = parAuthService.retrieveParams(uuid,
+                    request.getParameter(OAuthConstants.OAuth20Params.CLIENT_ID));
+            params.put(OAuthConstants.ALLOW_REQUEST_URI_AND_REQUEST_OBJECT_IN_REQUEST, "true");
+        } catch (ParCoreException e) {
+            throw new ParAuthFailureException(e.getMessage());
```

The actual exception is dropped. Throw that as well.

Since we are throwing a new exception from here let's have an error message like `Error occurred while retrieving params from PAR request` instead of `e. getMessage()`